### ISSUE TITLE
Docker input: add include/exclude filters for container names.

### ISSUE THREAD

**Issue by m4ce** [influxdata/telegraf#2734]
It would be handy if the docker input plugin supported (regex) patterns in container names.

```
## Only collect metrics for these containers, collect all if empty
  container_names = []
```

It'd be even nicer to have an include/exclude boolean option to drive the logic behind container_names.

Also, I noticed the plugin keeps reporting metrics for containers in exit state. Not sure what the reason is behind this, but maybe one can provide an option to disable this behavior?

**Comment by danielnelson** [comment]
The standard way we would do this is to have an include list and an exclude list, both would have glob patterns.  The same as for tagpass/tagdrop.

I'm open to regex but we need to decide on a nice way to expose them in a backwards compatible way, there was a little discussion about this yesterday on #2726.

>Also, I noticed the plugin keeps reporting metrics for containers in exit state. Not sure what the reason is behind this, but maybe one can provide an option to disable this behavior?

Seems like a nice feature, can you open a separate feature request for this?  I'm going to rename this issue for the first request.

**Comment by m4ce** [comment]
@danielnelsonm, sounds good.

One could add two more configuration parameters which should deprecate the container_names option in the future.

* include_container_names
* exclude_container_names.

I'd suggest aliasing container_names to include_container_names. If users are using container_names, each element can be turned into a regex like '^<name>$'. This will keep backwards compatibility.

**Comment by danielnelson** [comment]
Lets name them `container_name_include` and `container_name_exclude`?  For now lets only add glob support, I want to think about regex support a bit more.

**Comment by m4ce** [comment]
@danielnelson,

I will get started on this. Would you want to retain the current container_name option for backward compatibility?

**Comment by danielnelson** [comment]
Yeah, leave the current option in but document it as deprecated.  Thanks

**Comment by danielnelson** [comment]
Merged for 1.4.0
